### PR TITLE
Add USSD support to SMSD

### LINKED ISSUE #150: Add USSD support to SMSD [$200]
USSD is needed for example for querying prepaid card credit and would be useful addition to SMSD.
## 

<bountysource-plugin>

---
There is a **[$200 open bounty](https://www.bountysource.com/issues/28560918-add-ussd-support-to-smsd?utm_campaign=plugin&utm_content=tracker%2F253405&utm_medium=issues&utm_source=github)** on this issue. Add to the bounty at [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F253405&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by gnidorah**
Closes #150 
Thank you!

### CODE REVIEW

File: include/gammu-message.h [hunk 1]
```diff
@@ -436,6 +436,18 @@ typedef enum {
...
 /**
+ * SMS message class.
+ *
+ * \ingroup SMS
+ */
+typedef enum {
+	GSM_SMS_None = -1,
+	GSM_SMS_Flash,
+	GSM_SMS_Normal,
+	GSM_SMS_USSD
```
Comment (nijel) [inline include/gammu-message.h]: I'd prefer to use something out of usual range here. I know it's unlikely that there will be additional SMS class introduced, but as a safety measure, I'd choose something way off for USSD.

Comment (gnidorah) [reply include/gammu-message.h]: @nijel Thanks. Is that what you meant?

Comment (nijel) [reply include/gammu-message.h]: Yes, something like that (I was thinking about 255, but that really doesn't matter) ... probably the docs should be updated as well to match this.

Comment (gnidorah) [reply include/gammu-message.h]: @nijel Done, thank you!

Comment (gnidorah) [reply include/gammu-message.h]: @nijel Oops, it should be 127 because Class is unsigned char

**Comment by gnidorah** [comment]
@nijel Should I do anything else? Thanks

**Comment by nijel** [comment]
It would be great to have this part of USSD covered by tests, but I would understand if you skip this as SMSD testing is a bit tricky.

PS: I'm currently traveling, so I might not react as fast as usual.

**Comment by gnidorah** [comment]
@nijel Done, thanks. Have a good traveling!

**Comment by nijel** [comment]
Merged, thanks for your contribution!

**Comment by m7mdcc** [comment]
@nijel , is this published to your Repository 'launchpad ' ? 

Thanks

**Comment by nijel** [comment]
It's there, just the binaries for distros older than zesty didn't build for some reason. I have to find time to look at it...

**Comment by m7mdcc** [comment]
I'm on xenial , but when i send ussd request via gammu-smsd-inject USSD , it's sent . but i dont got anything on my inbox .. should i post issue ?

**Pull request merged**